components/sup/src/templating/context.rs

```diff
@@ -95,81 +112,805 @@ impl<'a> RenderContext<'a> {
+
+        map.serialize_entry("svc_path", &self.svc_path)?;
+        map.serialize_entry(
+            "svc_config_path",
+            &self.svc_config_path,
+        )?;
+        map.serialize_entry("svc_data_path", &self.svc_data_path)?;
+        map.serialize_entry("svc_files_path", &self.svc_files_path)?;
+        map.serialize_entry(
+            "svc_static_path",
+            &self.svc_static_path,
+        )?;
+        map.serialize_entry("svc_var_path", &self.svc_var_path)?;
+        map.serialize_entry("svc_pid_file", &self.svc_pid_file)?;
+        map.serialize_entry("svc_run", &self.svc_run)?;
+        map.serialize_entry("svc_user", &self.svc_user)?;
+        map.serialize_entry("svc_group", &self.svc_group)?;
+
+        map.end()
+    }
+}
+
+///////////////////////////////////////////////////////////////////////
+
+/// Templating proxy around a `census::CensusGroup`.
+///
+/// Currently exposed to users under the `svc` key.
+#[derive(Clone, Debug)]
+struct Svc<'a> {
+    service_group: Cow<'a, ServiceGroup>,
+    election_status: Cow<'a, ElectionStatus>,
+    update_election_status: Cow<'a, ElectionStatus>,
+    members: Cow<'a, Vec<SvcMember<'a>>>,
+    leader: Cow<'a, Option<SvcMember<'a>>>,
+    update_leader: Cow<'a, Option<SvcMember<'a>>>,
+    me: Cow<'a, SvcMember<'a>>,
+
+    // TODO (CM): this will need to be optional soon
+    first: SvcMember<'a>,
 }
 
 impl<'a> Svc<'a> {
+    // TODO (CM): rename to from_census_group
     fn new(census_group: &'a CensusGroup) -> Self {
         Svc {
-            service: census_group.service_group.service(),
-            group: census_group.service_group.group(),
-            org: census_group.service_group.org(),
-            election_is_running: census_group.election_status == ElectionStatus::ElectionInProgress,
-            election_is_no_quorum: census_group.election_status == ElectionStatus::ElectionNoQuorum,
-            election_is_finished: census_group.election_status == ElectionStatus::ElectionFinished,
-            update_election_is_running: census_group.election_status ==
-                ElectionStatus::ElectionInProgress,
-            update_election_is_no_quorum: census_group.election_status ==
-                ElectionStatus::ElectionNoQuorum,
-            update_election_is_finished: census_group.election_status ==
-                ElectionStatus::ElectionFinished,
-            me: SvcMember(census_group.me().expect("Missing 'me'")),
-            members: census_group
+            service_group: Cow::Borrowed(&census_group.service_group),
+            election_status: Cow::Borrowed(&census_group.election_status),
+            update_election_status: Cow::Borrowed(&census_group.update_election_status),
+            members: Cow::Owned(
+                census_group
+                    .members()
+                    .iter()
+                    .map(|m| SvcMember::from_census_member(m))
+                    .collect::<Vec<SvcMember<'a>>>(),
```

Comment:
This seems to build with just
```rust
                    .collect(),
```
Presumably the `from_census_member` is a strong enough hint for the type-inference to work. Is it valuable to be more explicit here, or would it be preferable to change to the simpler version?

Reply:
This code went through several iterations before I landed on the final implementation... one of those previous iterations required the type hint, but I guess the final one doesn't!

Thanks for catching this; I'll remove it.